src/utils/string/StringUtils.js

```diff
@@ -240,6 +242,117 @@ const getAnonymousChatName = async (members) => {
+  const members = [];
+  if (channel?.better_channel_member) {
+    if (isArray(channel?.better_channel_member)) {
+      _.forEach(channel?.better_channel_member, (member) => {
+        const isNotSelf = !helperIsSelf(member?.user_id, selfSignUserId, selfAnonUserId);
+        if (isNotSelf) {
+          members.push(member);
+        }
+      });
+
+      return members;
+    }
+
+    _.forIn(channel?.better_channel_member, (member, key) => {
+      const isNotSelf = !helperIsSelf(key, selfSignUserId, selfAnonUserId);
+      if (isNotSelf) {
+        members.push(member);
+      }
+    });
+
+    return members;
+  }
+
+  channel?.members?.forEach((item) => {
+    const isNotSelf = !helperIsSelf(item?.user_id, selfSignUserId, selfAnonUserId);
+    if (isNotSelf) {
+      members.push(item);
+    }
+  });
+
+  return members;
+};
+
+const getChannelListInfo = (channel, selfSignUserId, selfAnonUserId) => {
+  let channelName;
+  let channelImage;
+  let anonUserInfoEmojiCode;
+  let anonUserInfoEmojiName;
+  let anonUserInfoColorCode;
+  let anonUserInfoColorName;
+
+  const members = helperGetWhichMembers(channel, selfSignUserId, selfAnonUserId);
+
+  if (members?.length === 1) {
+    const member = members[0];
+    const isAnonymous = Boolean(member?.anon_user_info_emoji_name);
+
+    channelName = helperGetMemberName(member);
+
+    channelImage = isAnonymous ? DEFAULT_PROFILE_PIC_PATH : member?.user?.image;
+    anonUserInfoEmojiCode = member?.anon_user_info_emoji_code;
+    anonUserInfoEmojiName = member?.anon_user_info_emoji_name;
+    anonUserInfoColorCode = member?.anon_user_info_color_code;
+    anonUserInfoColorName = member?.anon_user_info_color_name;
+  } else if (members?.length > 6) {
+    const names = members?.slice(0, 6).map((item) => helperGetMemberName(item));
+    channelName = `${names.join(', ')} & others`;
+    channelImage = DEFAULT_PROFILE_PIC_PATH;
+  } else if (members?.length > 1) {
+    const names = members?.map((item) => helperGetMemberName(item));
+    channelName = names.join(', ');
+    channelImage = DEFAULT_PROFILE_PIC_PATH;
+  }
+
+  if (channel?.type === 'topics') {
+    channelImage = channel?.channel_image ?? DEFAULT_TOPIC_PIC_PATH;
+    channelName = channel?.name;
+  }
+
+  if (channel?.type === 'group') {
+    channelImage = channel?.is_custom_image ? channel?.channel_image : null;
+    if (channel?.is_custom_name) channelName = channel?.name;
+  }
+
+  return {
+    channelName,
+    channelImage,
+    anonUserInfoEmojiCode,
+    anonUserInfoEmojiName,
+    anonUserInfoColorCode,
+    anonUserInfoColorName
+  };
+};
+
+const getChannelMembers = (channel) => {
+  const members = helperGetWhichMembers(channel);
+
+  return members?.map((item) => {
+    const isAnonymous = Boolean(item?.anon_user_info_emoji_name);
+
+    return {
+      ...item,
+      name: helperGetMemberName(item),
+      image: isAnonymous ? DEFAULT_PROFILE_PIC_PATH : item?.user?.image
+    };
+  });
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `getChannelMembers` function, you're calling `helperGetWhichMembers` without passing the `selfSignUserId` and `selfAnonUserId` parameters, but the function definition requires these parameters. This might lead to unexpected behavior.
```diff
- const members = helperGetWhichMembers(channel);
+ const members = helperGetWhichMembers(channel, selfSignUserId, selfAnonUserId);
```
Please replace `selfSignUserId` and `selfAnonUserId` with the actual values.



<!-- This is an auto-generated comment by OSS CodeRabbit -->